src/consensus/aft/raft.h

```diff
@@ -1259,51 +1342,253 @@ namespace aft
+
+        switch (apply_success)
+        {
+          case kv::ApplyResult::FAIL:
+          {
+            LOG_FAIL_FMT("Follower failed to apply log entry: {}", i);
+            state->last_idx--;
+            ledger->truncate(state->last_idx);
+            send_append_entries_response(
+              r.from_node, AppendEntriesResponseType::FAIL);
+            break;
+          }
+
+          case kv::ApplyResult::PASS_SIGNATURE:
+          {
+            LOG_DEBUG_FMT("Deserialising signature at {}", i);
+            auto prev_lci = last_committable_index();
+            committable_indices.push_back(i);
+
+            if (ds->get_term())
+            {
+              // A signature for sig_term tells us that all transactions from
+              // the previous signature onwards (at least, if not further back)
+              // happened in sig_term. We reflect this in the history.
+              if (r.term_of_idx == aft::ViewHistory::InvalidView)
+              {
+                state->view_history.update(1, r.term);
+              }
+              else
+              {
+                state->view_history.update(prev_lci + 1, ds->get_term());
+              }
+              commit_if_possible(r.leader_commit_idx);
+            }
+            if (consensus_type == ConsensusType::BFT)
+            {
+              send_append_entries_signed_response(
+                r.from_node, ds->get_signature());
+            }
+            break;
+          }
+
+          case kv::ApplyResult::PASS:
+          {
+            break;
+          }
+
+          case kv::ApplyResult::PASS_SNAPSHOT_EVIDENCE:
+          {
+            break;
+          }
+
+          default:
+          {
+            throw std::logic_error("Unknown ApplyResult value");
+          }
+        }
+      }
+
+      return execute_append_entries_finish(confirm_evidence, r);
+    }
+
+    bool execute_append_entries_async(
+      std::unique_ptr<threading::Tmsg<AsyncExecution>>& msg)
+    {
+      // This function is responsible for selection the next batch of
+      // transactions we can execute concurrently and then starting said
+      // execution. If the are more pending entries after we select the batch we
+      // will return to this function.
+      std::vector<
+        std::tuple<std::unique_ptr<kv::AbstractExecutionWrapper>, kv::Version>>&
+        append_entries = msg->data.append_entries;
+      AppendEntries& r = msg->data.r;
+      bool confirm_evidence = msg->data.confirm_evidence;
+      async_exec_msg = std::move(msg);
+      bool is_first = true;
+      bool must_break = false;
+      uint64_t before_state_idx = state->last_idx;
+      bool run_sync = threading::ThreadMessaging::thread_count == 1;
+
+      while (async_exec_msg->data.index != append_entries.size())
+      {
+        auto& [ds, i] = append_entries[async_exec_msg->data.index];
+        if (!run_sync)
+        {
+          if (must_break)
```

Comment:
Can we document the return type of the call (it looks like false means we need to call again, which isn't necessarily what one might expect), and what must_break means (we've hit a serialisation point because of transaction dependencies, as far as I can tell?).